include/LightGBM/boosting.h

```diff
@@ -135,6 +136,17 @@ class LIGHTGBM_EXPORT Boosting {
     const double* features, double* output) const = 0;
 
   /*!
+  * \brief Prediction for one record, similar to PredictRaw but applies early stopping to speed up prediction.
+  *     In contrast to PredictRaw(), this function runs on a single thread.
+  *
+  * \param feature_values Feature value on this record
+  * \param output Prediction result for this record
+  * \param earlyStop Early stopping instance
+  */
+  virtual void PredictRawEarlyStop(const double* features, double* output,
+                            const PredictionEarlyStopInstance& earlyStop) const = 0;
```

Comment:
an early stopping struct instance is needed to predict a sample, and separates prediction early stopping from the configuration of GBDT, which I think helps keep the code an functionality separate.